gserver/src/server.ts

```diff
@@ -83,13 +83,13 @@ interface Page {
 
 //Return start, end position and matched (if any) Gherkin step
 function handleLine(line: String): StepLine {
-    let typeRegEx = /Given |When |Then |And /;
+    let typeRegEx = /Given |When |Then |And |But /;
     let typeMatch = line.match(typeRegEx);
     let typePart = typeMatch[0];
     let stepPart = line.replace(gerkinRegEx, '');
     let stepMatch;
     for (let i = 0; i < steps.length; i++) {
-        if (stepPart.search(steps[i].reg) !== -1) {
+        if (line.trim().match(steps[i].reg) || stepPart.search(steps[i].reg) !== -1) {

```

Comment:
Could you please provide valid example line, for which `stepPart.search(steps[i].reg) !== -1) ` will return false, but `line.trim().match(steps[i].reg)` will return true ?